### ISSUE TITLE
Wrong entries using exact search

### ISSUE THREAD

**Issue by rbleuse** [mvysny/aedict#951]
Hi Martin,

when I look for やつ in Exact search, I got the result 臣　やつこ which shouldn't be displayed as I don't use "Start with" search.
<img src="https://user-images.githubusercontent.com/626863/109248719-aad4d800-7829-11eb-8ed9-73bc23317a15.jpg" width="300" />

**Comment by rbleuse** [comment]
I think I understand the issue here, the search look for kanjis also (奴 is やつ) whereas my search is in **Dict** mode and not in **Kanjidict**

I reproduced the same issue with ひび (響き is found because 響 reading is ひび)
<img src="https://user-images.githubusercontent.com/626863/112593534-fbe8e380-8e4a-11eb-833b-f902e465d194.jpg" width="300" />

**Comment by mvysny** [comment]
Excellent catch, thank you. Let me take a look. Are you using the JMDict dictionary only, or also some other dictionary please?

**Comment by rbleuse** [comment]
Here is my dictionaries setup

<img src="https://user-images.githubusercontent.com/626863/113158961-343b5800-9277-11eb-9585-0a8a2e852f7b.jpg" width="300" />

**Comment by mvysny** [comment]
Thank you, I see that you're only using Jim Breen's JMDict and no other JMDict-compatible dictionaries (such as e.g. Aviation Dictionary nor Buddhism Dictionary). Let me test this please.

**Comment by mvysny** [comment]
Fixed, thanks for pointing it out! Fixed in Aedict 3.50.21

**Comment by rbleuse** [comment]
Hi Martin,

It seems that there are still some wrong results with the exact search :

<img src="https://user-images.githubusercontent.com/626863/124369707-cb1a1500-dca9-11eb-8b9c-c2a81c7c54d5.jpg" width="400" />

ですみ shouldn't be displayed

**Comment by rbleuse** [comment]
Another example with kanji :

<img src="https://user-images.githubusercontent.com/626863/124542149-cd24d500-de5d-11eb-85d2-c57c813a3ff8.jpg" width="300" />

But that one is probably because of special characters in the dictionary entry ?

**Comment by mvysny** [comment]
> But that one is probably because of special characters in the dictionary entry ?

Yeah that's probably it - I think I'm filtering those special characters out before the entry is indexed (mostly to get rid of braces and things like ~). But perhaps I should create an exception for these two particular characters.

Regardless, the ですみ should not be displayed regardless. I'll take a look.
